back/mysite3/views.py

```diff
@@ -11,18 +8,6 @@
 from rest_framework import viewsets
 from django.shortcuts import get_object_or_404
 
-
-@api_view(["get"])
-@permission_classes([AllowAny])
-def hello_world(request):
-    return Response({"message": "Hello, world!"})
-
-@api_view(["get"])
-@permission_classes([AllowAny])
-def q_list(request):
-    serializer = QuestionSeralizer(Question.objects.all())
-    return Response(serializer.data)
-
 @api_view(["get"])
 @permission_classes([AllowAny])
 def up_vote(request, choice_id):
```

Comment:
The `hello_world` example i gave you works for this and i did give it to you specifically knowing there was a more appropriate way to do this. I wanted you to se how you can do this outside viewsets too!!!! Consider a refactor to adding this to the viewset as an extra action then you'll inherit all of the serializers and such and it will be nice and clean and well organized. https://www.django-rest-framework.org/api-guide/viewsets/#marking-extra-actions-for-routing

Reply:
Thanks for the link. Now, I know how to do it both ways! See next commit for the extra action within the ChoiceViewset.